pytradfri/smart_task.py

```diff
@@ -0,0 +1,242 @@
+        """Return a list of keys for the given selection."""
+        return [k for k, b in self._lookup.items() if b & selection]
+
+    def get_selected_values(self, selection):
+        """Return a list of values for the given selection."""
+        return [v for b, v in self._choices if b & selection]
+
+
+WEEKDAYS = BitChoices(
+    (
+        ('mon', 'Monday'),
+        ('tue', 'Tuesday'),
+        ('wed', 'Wednesday'),
+        ('thu', 'Thursday'),
+        ('fri', 'Friday'),
+        ('sat', 'Saturday'),
+        ('sun', 'Sunday')
+    )
+)
+
+
+class SmartTask(object):
+    """Represent a group."""
+
+    def __init__(self, api, raw):
+        """Initialize smart task class."""
+        self.api = api
+        self.raw = raw
+
+    @property
+    def path(self):
+        """Return gateway path."""
+        return [ROOT_SMART_TASKS, self.id]
+
+    @property
+    def state(self):
+        """Boolean representing the light state of the transition."""
+        return self.raw.get(ATTR_LIGHT_STATE) == 1
+
+    @property
+    def created_at(self):
+        """Return when task was created."""
+        if ATTR_CREATED_AT not in self.raw:
+            return None
+        return datetime.utcfromtimestamp(self.raw[ATTR_CREATED_AT])
+
+    @property
+    def id(self):
+        """Return ID# of task."""
+        return self.raw.get(ATTR_ID)
+
+    @property
+    def task_type_id(self):
+        """Return type of task."""
+        return self.raw.get(ATTR_SMART_TASK_TYPE)
+
+    @property
+    def task_type_name(self):
+        """Return the task type in plain text.
+
+        (Own interpretation of names.)
+        """
+        if self.is_wake_up:
+            return "Wake Up"
+        if self.is_not_at_home:
+            return "Not At Home"
+        if self.if_lights_off:
+            return "Lights Off"
+
+    @property
+    def is_wake_up(self):
+        """Boolean representing if this is a wake up task."""
+        return self.raw.get(ATTR_SMART_TASK_TYPE) == ATTR_SMART_TASK_WAKE_UP
+
+    @property
+    def is_not_at_home(self):
+        """Boolean representing if this is a not home task."""
+        return self.raw.get(
+            ATTR_SMART_TASK_TYPE) == ATTR_SMART_TASK_NOT_AT_HOME
+
+    @property
+    def is_lights_off(self):
+        """Boolean representing if this is a lights off task."""
+        return self.raw.get(ATTR_SMART_TASK_TYPE) == ATTR_SMART_TASK_LIGHTS_OFF
+
+    @property
+    def repeat_days(self):
+        """Return int (bit) for enabled weekdays."""
+        return self.raw.get(ATTR_REPEAT_DAYS)
+
+    @property
+    def repeat_days_list(self):
+        """Binary representation of weekdays the event takes place."""
+        return WEEKDAYS.get_selected_values(self.raw.get(ATTR_REPEAT_DAYS))
+
+    @property
+    def start_action(self):
+        """Return state and all devices in smart action."""
+        return self.raw.get(ATTR_START_ACTION)
+
+    @property
+    def start_action_state(self):
```

Comment:
We should remove this property, people can access it via `task.start_action.state`